File: crates/varde-code/tests/fixtures/javascript/domain_specific.js
// Domain-specific fixtures: Express-style Route (`app.get("/path", ...)`) and
// Response (`res.status(200).json(...)`, `res.send(...)`, `res.sendStatus(n)`).

import express from "express";

const app = express();
const router = express.Router();

app.get("/users", (req, res) => {
  res.status(200).json({ users: [] });
});

app.post("/users", (req, res) => {
  res.status(201).json(req.body);
});

router.get("/health", (req, res) => {
  res.send("ok");
});

app.use("/auth", authRouter);

router.delete("/users/:id", (req, res) => {
  res.sendStatus(404);
});

File: crates/varde-code/tests/fixtures/python/domain_specific.py
# Domain-specific fixtures: Flask-style Route (`@app.route("/path")` decorator)
# and Response (`jsonify(...)` etc. — Flask's response-producing functions).

from flask import Flask, jsonify, render_template

app = Flask(__name__)


@app.route("/users")
def list_users():
    return jsonify([{"id": 1}])


@app.route("/users", methods=["POST"])
def create_user():
    return jsonify({"created": True})


@app.route("/health")
def health():
    return render_template("health.html")
